### PR TITLE
Determining whether a value is an array correctly

### PR DESCRIPTION
**Pull request by Wzb3422**
Using `input.length !== undefined` may cause wrong type judgment.
e.g.`const obj = { name: 'test', length: 1 };`

### CODE REVIEW

File: src/map.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export async function mapLimit<V>(input: any, limit: number, iteratee: any): Pro
     return [];
   }
 
-  const isArray = input.length !== undefined;
+  const isArray = input instanceof Array;
```
Comment (dhermes) [inline src/map.ts:76]: `Array.isArray()` exists

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray

Comment (ftrimble) [reply src/map.ts:76]: http://web.mit.edu/jwalden/www/isArray.html - this article was particularly interesting in regards to that. I tend to agree w/ @dhermes that if we're going to fix it, let's fix it whole hog.

Comment (Wzb3422) [reply src/map.ts:76]: I agree with you both, and we should fix the whole hog.

**Comment by gurpreetatwal** [comment]
#52 is somewhat related to this PR as well

**Pull request merged**